src/Microsoft.AspNetCore.Razor.Language/Extensions/DefaultTagHelperTargetExtension.cs

```diff
@@ -407,7 +407,18 @@ public void WriteTagHelperProperty(CodeRenderingContext context, DefaultTagHelpe
                             context.CodeWriter.WriteStartAssignment(GetPropertyAccessor(node));
                         }
 
-                        RenderTagHelperAttributeInline(context, node, node.Source);
+                        if (node.Children.Count == 0 &&
+                            node.AttributeStructure == AttributeStructure.Minimized &&
+                            (node.BoundAttribute.IsBooleanProperty ||
+                            (node.IsIndexerNameMatch && node.BoundAttribute.IsIndexerBooleanProperty)))
```

Comment:
Considering moving this and the line above to an extension method. Thoughts?

Reply:
You already do this logic in the intermediate node lowering phase; you can store that result on the `TagHelperPropertyIntermediateNode`. Kind of like how that node has the `IsIndexerNameMatch` today.

Reply:
That's the thing. It is not just done in IRL phase. It is also computed in the block rewriter.

Reply:
The block rewriter part is fine. It's done slightly differently. Lets do it in two places instead of three 😄. That's also consistent with how we pass information like this elsewhere